sound/soc/sof/loader.c

```diff
@@ -463,11 +528,34 @@ static int load_modules(struct snd_sof_dev *sdev, const struct firmware *fw)
 	return 0;
 }
 
+int strip_fw(struct snd_sof_dev *sdev, u32 bytes)
+{
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	struct firmware *stripped_fw;
+
+	stripped_fw = devm_kmalloc(sdev->dev, sizeof(*stripped_fw), GFP_KERNEL);
+
+	if (!stripped_fw)
+		return -ENOMEM;
+
+	memcpy(stripped_fw, plat_data->fw, sizeof(*stripped_fw));
+
+	stripped_fw->size -= bytes;
+	memmove((void *)&stripped_fw->data[0], &stripped_fw->data[bytes],
```

Comment:
just curious, why cant we adjust the size of plat_data->fw and do the memmove on it?

Reply:
I've done such a round around to omit `const` in `plat_data->fw` declaration, but casting to value without `const` will be much simpler - I thought there will be warning. Changed to simpler solution.

Reply:
When I don't move firmware content, then there will be problem with firmware loading -> `ext_man` is located at begin of firmware, and this part of binary must be dropped before `load_modules()`

Reply:
@ktrzcinx I understand, that other SOF code doesn't expect any extended manifest in the beginning of the firmware image, but cannot it be modified? I think it would be better to modify it, unless of course it's really a huge change, which shouldn't be the case.

Reply:
It's more clear and less error prone to just remove this part of content instead of looking for each place where it should be skipped. Especially when firmware load is platform specific part and extended manifest is generic.
Extended manifest is designed to by parsed during FW load routine and nowhere else, most effective way to fulfill this assumption is just dropping this content.

Reply:
This looks a bit iffy and I'm a bit concerned how this will pass upstream review. It does work though and seems safe w.r.t. release_firmware(). The fw->size will not reflect size of the allocation anymore, but it would seem this does no harm.

Reply:
So are you willing to create `fw_offset` field in `snd_sof_dev` and use it to skip this content in functions like `hda_dsp_cl_boot_firmware` ?

Reply:
...or just have a macro / function, that reads the size of the extended manifest and returns the beginning of the actual firmware - unless you have to calculate it, in which case yes, we can store that size in our data, I see no problem with that either.

Reply:
updated